### ISSUE TITLE
Upgrading to 9.22.4 aws s3 files say "You don't have permission to access this."

### ISSUE THREAD

**Issue by elplaza** [directus/directus#17279]
### Describe the Bug

My files are on AWS S3 and my configuration is like this:
```sh
STORAGE_LOCATIONS	aws
STORAGE_AWS_DRIVER	s3
STORAGE_AWS_KEY	        <my-aws-key>
STORAGE_AWS_SECRET	<my-aws-secret>
STORAGE_AWS_BUCKET	<my-aws-bucket>
STORAGE_AWS_REGION	<my-aws-region>
```

All was working fine with docker image 9.21.0.

Upgrading to 9.22.4 all files are "Forbidden" both for read and write operations.
UI network calls like:
`GET <directus>/assets/94f296a1-fcc7-468a-ba6b-da86dadc6395`
have 403 response with:
` {"errors":[{"message":"You don't have permission to access this.","extensions":{"code":"FORBIDDEN"}}]}`

### To Reproduce

Start with docker image 9.21.0 with storage configuration like this:
```sh
STORAGE_LOCATIONS	aws
STORAGE_AWS_DRIVER	s3
STORAGE_AWS_KEY	        <my-aws-key>
STORAGE_AWS_SECRET	<my-aws-secret>
STORAGE_AWS_BUCKET	<my-aws-bucket>
STORAGE_AWS_REGION	<my-aws-region>
```

Upgrade docker image to 9.22.4

### Errors Shown

UI network calls like:
`GET <directus>/assets/94f296a1-fcc7-468a-ba6b-da86dadc6395`
have 403 response with:
` {"errors":[{"message":"You don't have permission to access this.","extensions":{"code":"FORBIDDEN"}}]}`

### What version of Directus are you using?

9.22.4

### What version of Node.js are you using?

v18.13.0

### What database are you using?

MySQL

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

**Comment by paescuj** [comment]
It seems like you're missing the `STORAGE_<LOCATION>_ENDPOINT` configuration.
Please consolidate the [documentation](https://docs.directus.io/self-hosted/config-options.html#file-storage).
Also this issue might be helpful: #16876

Closing for now as it seems to be the result of a misconfiguration.

**Comment by RegisHubelia** [comment]
Hi - I have the configuration on my side - but I am getting the exact same error since upgrading to 9.22.4

**Comment by elplaza** [comment]
I added the STORAGE_AWS_ENDPOINT environment variable but I get always the same 403 error.

My configuration now is:
```sh
STORAGE_LOCATIONS	   aws
STORAGE_AWS_DRIVER	   s3
STORAGE_AWS_KEY	          <my-aws-key>
STORAGE_AWS_SECRET	  <my-aws-secret>
STORAGE_AWS_BUCKET	  <my-aws-bucket>/<my-aws-basedir>
STORAGE_AWS_REGION	  <my-aws-region>
STORAGE_AWS_ENDPOINT  <my-aws-endpoint>
```

I tried with all the following endpoint without luck:
- s3.amazonaws.com
- https://s3.amazonaws.com
- https://s3.<my-aws-region>.amazonaws.com
- <my-aws-bucket>.s3.<my-aws-region>.amazonaws.com
- https://<my-aws-bucket>.s3.<my-aws-region>.amazonaws.com
- https://<my-aws-bucket>.s3.<my-aws-region>.amazonaws.com/

This env var was not required, now it is? Why?
It would be great if you tell me where can i find this endpoint on aws.
Sorry but reading docs it is not so clear what endpoint are you talking about . . .
